File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        long l = 0;
        long r = floor(pow(c, 0.5));
        while (l <= r){
            long total = pow(l, 2) + pow(r, 2);
            if (total == c){
                return true;
            }else if(total < c){
                l++;
            }else{
                r--;
            }
        }
        return false;
    }
};

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        l = 0
        r = int(c**0.5)
        while l <= r:
            total = l ** 2 + r ** 2
            if total == c:
                return True
            elif total > c:
                r -= 1
            elif total < c:
                l += 1
            
        return False